os/drivers/testcase/test_group.c

```diff
@@ -0,0 +1,256 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <debug.h>
+#include <errno.h>
+#include <sched.h>
+#include <sys/types.h>
+#include <tinyara/testcase_drv.h>
+#include <tinyara/sched.h>
+#ifdef CONFIG_SCHED_HAVE_PARENT
+#include <group/group.h>
+#ifdef CONFIG_SCHED_CHILD_STATUS
+#include <task/task.h>
+#endif
+#endif
+#include "kernel_test_proto.h"
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#if defined(CONFIG_SCHED_HAVE_PARENT) && defined(CONFIG_SCHED_CHILD_STATUS)
+#define TASK_STACKSIZE 2048
+static struct tcb_s *st_tcb;
+static struct task_group_s *group;
+static struct child_status_s *child;
+static struct child_status_s *child_returned;
+static pid_t child_pid;
+#endif
+
+/****************************************************************************
+ * Private Function
+ ****************************************************************************/
+
+#if defined(CONFIG_SCHED_HAVE_PARENT) && defined(CONFIG_SCHED_CHILD_STATUS)
+static int test_sched_self(void)
+{
+	st_tcb = sched_self();
+	if (st_tcb == NULL) {
+		dbg("sched_self failed.");
+		return ERROR;
+	}
+
+	group = st_tcb->group;
+	if (group == NULL) {
+		dbg("group is null.");
+		return ERROR;
+	}
+
+	return OK;
+}
+
+static int group_exitchild_func(int argc, char *argv[])
+{
+	task_delete(0);
+	return ERROR;
+}
+
+static int test_group_exit_child(int cmd, unsigned long arg)
+{
+	int ret;
+	ret = test_sched_self();
+	if (ret == ERROR) {
+		return ERROR;
+	}
+	child_pid = kernel_thread("group", SCHED_PRIORITY_DEFAULT, TASK_STACKSIZE, group_exitchild_func, (char *const *)NULL);
+	if (child_pid < 0) {
+		dbg("task_create failed.");
+		return ERROR;
+	}
+
+	child = group_findchild(group, child_pid);
+	if (child == NULL) {
+		dbg("child is null.");
+		return ERROR;
+	}
+
+	sleep(3);
```

Comment:
Why do you need 3 secs? I think 1 us is enough.

Reply:
I agree.
I will change sleep(3) to usleep(1)